benchmark/url/urlpattern-parse.js

```diff
@@ -0,0 +1,27 @@
+'use strict';
+const common = require('../common.js');
+const { URLPattern } = require('url');
+const { ok } = require('assert');
+
+const tests = [
+  'https://(sub.)?example(.com/)foo',
+  { 'hostname': 'xn--caf-dma.com' },
+  { 'pathname': '/foo', 'search': 'bar', 'hash': 'baz',
+    'baseURL': 'https://example.com:8080' },
+  { 'pathname': '/([[a-z]--a])' },
+];
+
+const bench = common.createBenchmark(main, {
+  pattern: tests.map(JSON.stringify),
+  n: [1e5],
+});
+
+function main({ pattern, n }) {
+  const inputPattern = JSON.parse(pattern);
+  bench.start();
+  for (let i = 0; i < n; i += 1) {
+    const p = new URLPattern(inputPattern);
+    ok(p);
+  }
+  bench.end(n);
```

Comment:
```suggestion
  let deadCode = null;
  bench.start();
  for (let i = 0; i < n; i += 1) {
    deadCode = new URLPattern(inputPattern);
  }
  bench.end(n);
  ok(deadCode);
```

You can leave the `ok` out to avoid perf issues related to the assert ok.